content/en/blog/_posts/2020-12-07-Snapshot-Restore-Volume-Support-for-Kubernetes-CRD-External-Controller.md

```diff
@@ -0,0 +1,217 @@
+
+## Why add Volume Snapshots to Kubernetes?
+
+Kubernetes aims to create an abstraction layer between distributed applications and underlying clusters so that applications can be agnostic to the specifics of the cluster they run on and application deployment requires no “cluster-specific” knowledge.
+
+The Kubernetes Storage SIG identified snapshot operations as critical functionality for many stateful workloads. For example, a database administrator may want to snapshot a database’s volumes before starting a database operation.
+
+By providing a standard way to trigger volume snapshot operations in Kubernetes, this feature allows Kubernetes users to incorporate snapshot operations in a portable manner on any Kubernetes environment regardless of the underlying storage.
+
+Additionally, these Kubernetes snapshot primitives act as basic building blocks that unlock the ability to develop advanced enterprise-grade storage administration features for Kubernetes,  including application or cluster level backup solutions.
+
+## What’s new since beta?
+
+With the promotion of Volume Snapshot to GA, the feature is enabled by default on standard Kubernetes deployments and cannot be turned off.
+
+Many enhancements have been made to improve the quality of this feature and to make it production-grade.
+
+The Volume Snapshot APIs and client library were moved to a separate go module.
+A snapshot validation webhook has been added to perform necessary validation on volume snapshot objects. More details can be found [here](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1900-volume-snapshot-validation-webhook).
+
+Along with the validation webhook, the volume snapshot controller will start labeling invalid snapshot objects which already existed. This allows users to identify, remove any invalid objects, and correct their workflows. Once the API is switched to the v1 type, those invalid objects will not be deletable from the system.
+To provide better insights how the snapshot feature is performing, an initial set of operation metrics has been added to the volume snapshot controller.
+
+More e2e tests have been added to test the feature in a real Kubernetes cluster on GCP. Stress tests (based on Google Persistent Disk and hostpath CSI Drivers) have been introduced to test the robustness of the system.
+
+Other than introducing tightening validation, there is no difference between the v1beta1 and v1 Kubernetes volume snapshot API. In this release (with Kubernetes 1.20), both v1 and v1beta1 are served while the stored API version is still v1beta1. Future releases will switch the stored version to v1 and gradually remove v1beta1 support.
+
+## Which CSI Drivers support Volume Snapshots?
+
+Snapshots are only supported for CSI drivers, not for in-tree or Flex. Ensure the deployed CSI driver on your cluster has implemented the snapshot interfaces. For more information, see Container Storage Interface (CSI) for Kubernetes GA.
```

Comment:
If required, please provide the link. Would be happy to add.  I think it would be a value add.

Reply:
The blog link is here: https://kubernetes.io/blog/2019/01/15/container-storage-interface-ga/

Reply:
Addressed this in the latest commit, PTAL.

> Some suggestions 👍 Mostly non-blocking nits - if you could though, please remove the spaces from the file name and replace with a `-`. If possible, we like to avoid spaces in filenames^^;;;

@mrbobbytables : Have hyphenated the filename in the latest commit & almost all of the suggestions made by you & @sftim have been incorporated. PTAL.